src/RefractDataStructure.cc

```diff
@@ -10,8 +10,50 @@
 #include "RefractDataStructure.h"
 #include "refract/AppendDecorator.h"
 
+#include "RefractSourceMap.h"
+
 namespace drafter {
 
+    template <typename T, typename V = typename T::ValueType>
+    struct Append {
+        typedef T ElementType;
+        typedef V ValueType;
+        ElementType*& element;
+
+        Append(ElementType*& e) : element(e)
+        {
+        }
+
+        void operator()(const NodeInfo<ValueType>& value)
+        {
+            //throw std::logic_error("Can not append to primitive type");
```

Comment:
Ok will change as you suggest. And swap position explaining comment with commented line. We discuss more times same topic.